### PR TITLE
Setup correct Syslog hooks for trace Logger

### PR DESCRIPTION
**Pull request by lcastellano**
Syslog hooks are currently not properly setup for the trace Logger. That causes the operation syslog messages to go lost.

Fixes 7169

### CODE REVIEW

File: pkg/trace/trace.go [hunk 1]
```diff
@@ -66,6 +66,15 @@ func (t *Message) delta() time.Duration {
 	return time.Now().Sub(t.startTime)
 }
 
+// Add Syslog hook
+func InitLogger(cfg *log.LoggingConfig) error {
+	hook, err := log.GetSyslogHook(cfg)
+	if err == nil && hook != nil {
+		Logger.Hooks.Add(hook)
```
Comment (hickeng) [inline pkg/trace/trace.go]: I suspect we'll get away with this but if I recall correctly `Hooks.Add` is not thread safe.
Please confirm I'm correctly, that there's no risk of calling this method concurrently and note thread safety in the function comment.

**Comment by mhagen-vmware** [comment]
this really should have some unit tests, but I know that we have an integration test already for this - thus the lgtm

File: pkg/log/log.go [hunk 2]
```diff
@@ -70,22 +70,28 @@ func Init(cfg *LoggingConfig) error {
...
+			hook, err := GetSyslogHook(cfg)
+			if err == nil && hook != nil {
+				logrus.AddHook(hook)
 			}
 		})
 
 	return initializer.err
 }
+
+func GetSyslogHook(cfg *LoggingConfig) (logrus.Hook, error) {
```
Comment (hmahmood) [inline pkg/log/log.go]: Nit: `CreateSyslogHook` would be closer to what this function is doing.

**Comment by anchal-agrawal** [comment]
@lcastellano The commit message itself still contains `...: issue 7169`. I've noticed that Github often uses the commit message (not the PR title) while merging. It'd be great if we updated the commit too.

**Pull request merged**